test/extended/util/framework.go

```diff
@@ -280,3 +281,19 @@ func ExtendedTestPath() string {
 func FixturePath(elem ...string) string {
 	return filepath.Join(append([]string{ExtendedTestPath()}, elem...)...)
 }
+
+// CheckURL grabs the output from the specified url and returns it.
+// It will retry up to N times if an error occurs during the request.
+func CheckURL(url string, retries int) (response string, err error) {
+	r, err := http.Get(url)
+	count := 0
+	for count < retries && err != nil {
```

Comment:
```
err := wait.Poll(time.Second, time.Minute, func() (bool, error) {
  content, err := doYourThing()
  if err != nil {
    t.Logf("your error: %v", err)
  }
  if content != expected {
    return false, nil
  }
  return true, nil
})
if err != nil {
  timeout
}
```

Is 5 lines of code, uses wait poll, and the efficiency argument in test code is a bit suspect (you can just poll faster).